portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,23 +51,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Post> posts = entityToList(results, request);
     
     response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
     response.getWriter().println(convertToJson(posts));
   }
   
   // Posts information to the server to save as inputs from the data form
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity commentEntity = createEntity(request);
-
+  
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/contact.html");
   }
+  
+  private String getLanguage(HttpServletRequest request) {
+    String languageChoice;
+    try {
+      languageChoice = request.getParameter("language");
+    } catch (NullPointerException e) {
```

Comment:
It's a good practice to log caught exceptions with a message indicating what may have gone wrong. That way you're not blind to failures like this. It may also help debugging other issues.

Reply:
Done! Let me know if you think there is a better way to log errors.

Reply:
You may want to use the java logger instead. Here is a starting reference: https://www.loggly.com/ultimate-guide/java-logging-basics/ 

And here is a stackoverflow link with more context why: https://stackoverflow.com/questions/31869391/what-is-the-difference-between-java-logger-and-system-out-println/31869411#:~:text=Logger%20is%20useful%20when%20you,you%20cannot%20check%20the%20console.&text=println%20always%20logs%20the%20message,in%20the%20logger%20configuration%20file.

Reply:
Okay, this is good to know! I added a logger instance to log all of the errors